src/Module/FilterCartItemPriceHandler.php

```diff
@@ -0,0 +1,246 @@
+     * @param array|ArrayAccess|ContainerInterface|stdClass $cartItemConfig    The cart item configuration.
+     */
+    public function __construct(
+        SelectCapableInterface $bookingsSelectRm,
+        EvaluableInterface $priceEvaluator,
+        FactoryInterface $valueAwareFactory,
+        $exprBuilder,
+        $cartItemConfig
+    ) {
+        $this->_setPriceEvaluator($priceEvaluator);
+        $this->_setBookingValueAwareFactory($valueAwareFactory);
+
+        $this->bookingsSelectRm = $bookingsSelectRm;
+        $this->exprBuilder      = $exprBuilder;
+        $this->cartItemConfig   = $cartItemConfig;
+
+    }
+
+    /**
+     * Retrieves the booking price evaluator.
+     *
+     * @since [*next-version*]
+     *
+     * @return EvaluableInterface|null The price evaluator instance, if any.
+     */
+    protected function _getPriceEvaluator()
+    {
+        return $this->priceEvaluator;
+    }
+
+    /**
+     * Sets the booking price evaluator.
+     *
+     * @since [*next-version*]
+     *
+     * @param EvaluableInterface|null $priceEvaluator The price evaluator instance, if any.
+     *
+     * @throws InvalidArgumentException If the argument is not an evaluable instance.
+     */
+    protected function _setPriceEvaluator($priceEvaluator)
+    {
+        if ($priceEvaluator !== null && !($priceEvaluator instanceof EvaluableInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an evaluable instance'), null, null, $priceEvaluator
+            );
+        }
+
+        $this->priceEvaluator = $priceEvaluator;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     *
+     * @return void
+     */
+    public function __invoke()
+    {
+        $event = func_get_arg(0);
+
+        if (!($event instanceof EventInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an event instance'), null, null, $event
+            );
+        }
+
+        $options = $event->getParam(2);
+
+        $eddBkKey      = $this->_containerGetPath($this->cartItemConfig, ['data', 'eddbk_key']);
+        $bookingIdKey  = $this->_containerGetPath($this->cartItemConfig, ['data', 'booking_id_key']);
+        $bookingIdPath = [$eddBkKey, $bookingIdKey];
+
+        try {
+            $bookingId = $this->_containerGetPath($options, $bookingIdPath);
+        } catch (NotFoundExceptionInterface $exception) {
+            return;
+        }
+
+        // Alias expression builder
+        $b = $this->exprBuilder;
+
+        // Fetch the corresponding booking from storage
+        $condition = $b->eq($b->var('id'), $b->lit($bookingId));
+        $bookings  = $this->bookingsSelectRm->select($condition);
+        // Stop if no booking was found, or if multiple bookings matched the ID (for some reason?)
+        if ($this->_countIterable($bookings) !== 1) {
+            return;
```

Comment:
If no booking is found - does that mean that simply no booking corresponds to it, and it should be left untouched?

Reply:
Yeah